clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/spark/datawriter/jobs/DataWriterSparkJob.java

```diff
@@ -29,30 +31,59 @@ protected Dataset<Row> getUserInputDataFrame() {
     SparkSession sparkSession = getSparkSession();
     PushJobSetting pushJobSetting = getPushJobSetting();
 
-    DataFrameReader dataFrameReader = sparkSession.read();
-    dataFrameReader.format(VeniceHdfsSource.class.getCanonicalName());
-    setInputConf(sparkSession, dataFrameReader, INPUT_PATH_PROP, new Path(pushJobSetting.inputURI).toString());
-    setInputConf(sparkSession, dataFrameReader, KEY_FIELD_PROP, pushJobSetting.keyField);
-    setInputConf(sparkSession, dataFrameReader, VALUE_FIELD_PROP, pushJobSetting.valueField);
-    if (pushJobSetting.etlValueSchemaTransformation != null) {
-      setInputConf(
-          sparkSession,
-          dataFrameReader,
-          ETL_VALUE_SCHEMA_TRANSFORMATION,
-          pushJobSetting.etlValueSchemaTransformation.name());
-    }
     if (pushJobSetting.isAvro) {
-      setInputConf(sparkSession, dataFrameReader, SCHEMA_STRING_PROP, pushJobSetting.inputDataSchemaString);
-      if (pushJobSetting.generatePartialUpdateRecordFromInput) {
-        setInputConf(sparkSession, dataFrameReader, GENERATE_PARTIAL_UPDATE_RECORD_FROM_INPUT, String.valueOf(true));
-        setInputConf(sparkSession, dataFrameReader, UPDATE_SCHEMA_STRING_PROP, pushJobSetting.valueSchemaString);
-      }
-      setInputConf(sparkSession, dataFrameReader, VSON_PUSH, String.valueOf(false));
+      return getAvroDataFrame(sparkSession, pushJobSetting);
     } else {
-      setInputConf(sparkSession, dataFrameReader, VSON_PUSH, String.valueOf(true));
-      setInputConf(sparkSession, dataFrameReader, FILE_KEY_SCHEMA, pushJobSetting.vsonInputKeySchemaString);
-      setInputConf(sparkSession, dataFrameReader, FILE_VALUE_SCHEMA, pushJobSetting.vsonInputValueSchemaString);
+      return getVsonDataFrame(sparkSession, pushJobSetting);
     }
-    return dataFrameReader.load();
+  }
+
+  private Dataset<Row> getAvroDataFrame(SparkSession sparkSession, PushJobSetting pushJobSetting) {
+    Dataset<Row> df =
+        sparkSession.read().format("avro").option("pathGlobFilter", GLOB_FILTER_PATTERN).load(pushJobSetting.inputURI);
+
+    // Transforming the input data format
+    df = df.map((MapFunction<Row, Row>) (Row record) -> {
```

Comment:
Had no idea this syntax was possible for a map closure...!

Reply:
Even I didn't realize this `(Row record)` implementation. Apparently, it is not necessary to have the type. Reliance on IDEs has made us dumber.

The `(MapFunction<Row, Row>)` is a type casting of the lambda, that we need to do because of some Scala/Java weirdness